### ISSUE TITLE
Layout.getDesiredWidth() cannot measure the width of emojis on Android <=4.4.2

### ISSUE THREAD

**Issue by zongjingyao** [facebook/react-native#11592]
```
render() {
    return (
      <View style={{flex: 1, alignItems: 'flex-start', flexDirection: 'column'}}>
        <Text style={{backgroundColor: 'green', color: 'black', fontSize: 18}}>
          😀
        </Text>
      </View>
    );
}
```
This is my code. No problem on Android devices >=4.4.4 and iOS devices. But on Android devices <= 4.4.2, this is nothing on the screen.
I find that React Native uses Layout.getDesiredWidth() to measure the width of Text in [ReactTextShadowNode](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java#L242). But the value returned from Layout.getDesiredWidth() is always 0 on Android devices <= 4.4.2, which means Android(<=4.4.2) cannot measure the width of emojis by Layout.getDesiredWidth(). So it should be a bug of Android OS(<=4.4.2). I don’t know how to fix this, any workaround is welcome. Please help me, thanks in advance!

**Comment by wuhenwuya** [comment]
@zongjingyao I also encountered the same problem,how did you solve it ? And How to be compatible with emoji (ios 、android)?

**Comment by zongjingyao** [comment]
Hi @wuhenwuya , the problem is still there, I don't how to fix it.
You want to have same emoji on both iOS and Android?

**Comment by wuhenwuya** [comment]
@zongjingyao ,是的，android4.4.2获取不到发送的emoji长度（js emoji.length可以获取到，react native底层获取不到），还有android ios emoji的兼容也没有找到好的方法

**Comment by zongjingyao** [comment]
@wuhenwuya 兼容的话基本上都是在Android上显示iOS的emoji，你可以看一下下面这两个链接。
http://ragnraok.github.io/android-emoji-font-method.html
https://github.com/w446108264/XhsEmoticonsKeyboard

**Comment by wuhenwuya** [comment]
@facebook-github-bot The issues has been solved, but the code has not changed in ReactNative43(react-native init or [https://github.com/facebook/react-native/archive/v0.43.1.zip](url) [https://github.com/facebook/react-native/archive/v0.43.3.zip](url))

**Comment by zongjingyao** [comment]
Hi @wuhenwuya The code has been merged into v0.44.
https://github.com/facebook/react-native/blob/0.44-stable/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java#L87
